freeipa.spec.in

```diff
@@ -1081,6 +1081,19 @@ fi
 
 %if ! %{ONLY_CLIENT}
 
+%triggerin server -- tomcat >= 9.0.31
+%{__python3} -c "import sys; from ipaserver.install import installutils; sys.exit(0 if installutils.is_ipa_configured() else 1);" > /dev/null 2>&1
+if [  $? -eq 0 ]; then
+    # Run upgrade code if new tomcat is installed but AJP connector is not
+    # encrypted yet. Use systemd-run to create an isolated environment to
+    # detach from the security context used by the rpm transaction
+    [ -f /etc/httpd/conf.d/ipa-pki-proxy.conf ] && \
+    grep -q secret= /etc/httpd/conf.d/ipa-pki-proxy.conf || \
+    /usr/bin/systemd-run --description "Upgrade FreeIPA to secure AJP connector for tomcat 9.0.31+" \
+                         --unit=ipa-server-upgrade-ajp-connector --wait \
+                         /usr/sbin/ipa-server-upgrade --quiet
```

Comment:
@abbra, if I understand correctly you want to trigger upgrade only if there is `ipa-pki-proxy.conf` **and** the latter doesn't contain `secret`. But the current code reflects a differ logic and it will trigger upgrade in two cases: not existed file `ipa-pki-proxy.conf` **or** missing `secret`.

Reply:
Yes, I need to wrap the second check into full if..fi statement instead.